clinica/iotools/bids_utils.py

```diff
@@ -257,13 +255,24 @@ def create_sessions_dict_OASIS(
                     session_names = get_bids_subjs_list(subj_dir)
                     for s in session_names:
                         s_name = s.replace("ses-", "")
+                        if study_name != "OASIS3":
+                            row = file_to_read.iloc[r]
+                        else:
+                            mr_id = subj_id + "_ClinicalData_" + s_name
+                            row = file_to_read[file_to_read["MR ID"] == mr_id].iloc[0]
                         if subj_bids not in sessions_dict:
                             sessions_dict.update({subj_bids: {}})
                         if s_name not in sessions_dict[subj_bids].keys():
                             sessions_dict[subj_bids].update({s_name: {"session_id": s}})
                         (sessions_dict[subj_bids][s_name]).update(
                             {sessions_fields_bids[i]: row[sessions_fields[i]]}
                         )
+                        # Calculate the difference in months for OASIS3 only
+                        if study_name == "OASIS3" and sessions_fields_bids[i] == "age":
+                            diff_years = float(sessions_dict[subj_bids][s_name]["age"]) - participants_df[participants_df["participant_id"] == subj_bids]["age_bl"]
+                            (sessions_dict[subj_bids][s_name]).update(
+                                {"diff_months": round(float(diff_years) * 12)}
+                            )
 
     return sessions_dict
```

Comment:
When testing oasis3, the converter crashes when it calls for session_dict. It appears to be empty when I test it.

Reply:
Please, test with the latest version of the OASIS3 data-test and tell me again if it failed : https://owncloud.icm-institute.org/index.php/s/qMoXZIoA3CgawJX

Reply:
If it failed again, could you copy paste a screenshot of the error ?

Reply:
It works out with the latest oasis3 data test, thanks !